jss-sparsegl/script_for_plotting.R

```diff
@@ -0,0 +1,134 @@
+#                    rep(c(-5, 0, 1, 5), each = 5, times = (p - 20) / 20))
+#   } else {
+#     beta_star <- c(rep(5, 5), c(5, -5, 2, 0, 0), rep(-5, 5), c(2, -3, 8, 0, 0),
+#                    rep(c(rep(5, 5), rep(0, 5), rep(-5, 5), c(2, 3, 5, 0, 0)), times = (p - 20) / 20))
+#   }
+#   beta_star
+# }
+# 
+# 
+# simulate <- function(change_list,
+#                      change_feature = c("n", "p"),
+#                      within_group = TRUE,
+#                      group_size = 5,
+#                      family = c("gaussian", "binomial")) {
+#   change_feature <- match.arg(change_feature)
+#   family <- match.arg(family, c("gaussian", "binomial"))
+#   df <- list()
+#   k <- 1
+# 
+#   if (change_feature == "n") {
+#     p <- 100
+#     beta_star <- beta_generate(p, within_group, group_size)
+#     groups <- rep(1:(p / group_size), each = group_size)
+#   } else {
+#     n <- 100
+#   }
+# 
+#   for (i in seq(length(change_list))) {
+#     if (change_feature == "n") {
+#       n <- change_list[i]
+#     } else {
+#       p <- change_list[i]
+#       beta_star <- beta_generate(p, within_group, group_size)
+#       groups <- rep(1:(p / group_size), each = group_size)
+#     }
+#     set.seed(101)
+#     X <- matrix(rnorm(n * p), nrow = n)
+#     for (j in seq(50)) {
+#       set.seed(j)
+#       if (family == "gaussian") {
+#         eps <- rnorm(n)
+#         y <- X %*% beta_star + eps
+#         data <- list(x = X, y = y)
+#         y_gglasso <- y
+#       } else {
+#         pr <- 1 / (1 + exp(-X %*% beta_star))
+#         y <- rbinom(n, 1, pr)
+#         data <- list(x = X, y = y)
+#         y_gglasso <- y
+#         y_gglasso[which(y_gglasso == 0)] <- -1
+#       }
+#       if (!within_group) {
+#         df[[k]] <- as.data.frame(summary(
+#           microbenchmark(sparsegl = sparsegl(X, y, groups, family = family, asparse = 0, standardize = FALSE),
+#                          SGL = SGL(data, groups, type = ifelse(family == "gaussian", "linear", "logit"), alpha = 0, nlam = 100, standardize = FALSE),
+#                          gglasso = gglasso(X, y_gglasso, groups, loss = ifelse(family == "gaussian", "ls", "logit")),
+#                          times = 1)))
+#       } else {
+#         df[[k]] <- as.data.frame(summary(
+#           microbenchmark(sparsegl = sparsegl(X, y, groups, family = family, asparse = 0.05, standardize = FALSE),
+#                          SGL = SGL(data, groups, type = ifelse(family == "gaussian", "linear", "logit"), alpha = 0.05, nlam = 100, standardize = FALSE),
+#                          times = 1)))
+#       }
+#       df[[k]] <- df[[k]] %>%
+#         select(expr, mean) %>%
+#         mutate(
+#           change_feature = factor(change_feature),
+#           change_list := factor(!!rlang::sym(change_feature)),
+#           within_group = within_group)
+#       k <- k + 1
+#       print(k)
+#     }
+#   }
+#   datf <- bind_rows(df)
+#   return(datf)
+# }
+
+
+plot_box <- function(df) {
```

Comment:
1. Move this to a separate script.
2. Put it in a chunk in the .Rmd with `eval = TRUE`

Reply:
"move this to a separate script": move into the same .Rmd as what the last comment pointed to with eval = TRUE or another new .Rmd?

Reply:
* Functions should live in their own script.
* Calls to those functions can live in the .Rmd (if they run in reasonable time), or otherwise in a separate script that is run once, and the output saved. Then the saved output can be loaded.

Reply:
Do you mean .Rmd by "jss-sparsegl.Rmd"?

Reply:
I mean, move the `plot_box()` function (and other functions) to an `.R` script and source that script in `jss-sparsegl.Rmd`. You could name it `helper-functions.R`. That way, it doesn't take up huge amounts of space in the document. Then in `jss-sparsegl.Rmd` you source the script in a chunk. And later on can call `plot_box()` to produce the graphics.